### ISSUE TITLE
Database: index sequence on table aow_processed does not comply with the query being fired.

### ISSUE THREAD

**Issue by neerajcrespo** [salesagility/SuiteCRM#2969]
#### Issue
Select Query on aow_processed does not fully uses the index 'aow_processed_index_workflow' on the same table because of sequence mismatch.

Following is the index sequence.
```
mysql> show index from aow_processed;
+---------------+------------+---------------------------------+--------------+-----------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| Table         | Non_unique | Key_name                        | Seq_in_index | Column_name     | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment |
+---------------+------------+---------------------------------+--------------+-----------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| aow_processed |          0 | PRIMARY                         |            1 | id              | A         |     1582407 |     NULL | NULL   |      | BTREE      |         |               |
| aow_processed |          1 | aow_processed_index_workflow    |            1 | aow_workflow_id | A         |          18 |     NULL | NULL   | YES  | BTREE      |         |               |
| aow_processed |          1 | aow_processed_index_workflow    |            2 | status          | A         |          18 |     NULL | NULL   | YES  | BTREE      |         |               |
| aow_processed |          1 | aow_processed_index_workflow    |            3 | parent_id       | A         |      395601 |     NULL | NULL   | YES  | BTREE      |         |               |
| aow_processed |          1 | aow_processed_index_workflow    |            4 | deleted         | A         |      395601 |     NULL | NULL   | YES  | BTREE      |         |               |
| aow_processed |          1 | aow_processed_index_status      |            1 | status          | A         |          18 |     NULL | NULL   | YES  | BTREE      |         |               |
| aow_processed |          1 | aow_processed_index_workflow_id |            1 | aow_workflow_id | A         |          18 |     NULL | NULL   | YES  | BTREE      |         |               |
+---------------+------------+---------------------------------+--------------+-----------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
7 rows in set (0.10 sec)
```

The queries which got slowdown is 
`SELECT * FROM aow_processed WHERE aow_processed.aow_workflow_id=? AND aow_processed.parent_id=leads.id AND aow_processed.status = ? AND aow_processed.deleted = ?`

`SELECT aow_processed.* FROM aow_processed  WHERE aow_workflow_id = ? AND parent_id = ? AND deleted=? LIMIT ?,?`

#### Possible Fix
change the sequence in where clause to match with index sequence or create new index for these queries.

#### Context
This is accounting for performance hit on DB where ever this table is being read.

#### Your Environment

* SuiteCRM Version used:
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version (e.g. MySQL, PHP 7):
* Operating System and version (e.g Ubuntu 16.04):

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Database: index sequence on table aow_processed does not comply with the query being fired.](https://trello.com/c/yE4B61Zu/40-database-index-sequence-on-table-aow-processed-does-not-comply-with-the-query-being-fired)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
